modules/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangPackageBuilder.java

```diff
@@ -205,30 +208,34 @@ public void addValueType(DiagnosticPos pos, Set<Whitespace> ws, String typeName)
         addType(typeNode);
     }
 
-    public void addArrayType(DiagnosticPos pos, int dimensions) {
+    public void addArrayType(DiagnosticPos pos, Set<Whitespace> ws, int dimensions) {
         BLangType eType = (BLangType) this.typeNodeStack.pop();
         BLangArrayType arrayTypeNode = (BLangArrayType) TreeBuilder.createArrayTypeNode();
+        arrayTypeNode.addWS(ws);
         arrayTypeNode.pos = pos;
         arrayTypeNode.elemtype = eType;
         arrayTypeNode.dimensions = dimensions;
 
         addType(arrayTypeNode);
     }
 
-    public void addUserDefineType() {
+    public void addUserDefineType(Set<Whitespace> ws) {
```

Comment:
Can you update this with the position, if it is not set already

Reply:
It's already set. 

https://github.com/ballerinalang/ballerina/blob/7c0bcae6fc520797d12c10ba667a1689ea8dc539/modules/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangPackageBuilder.java#L286

Reason I am accumulating WS at each Node modification is that for some Nodes, WS is coming form multiple places (Eg: Function WS= CallableUnitBody WS + FunctionDef WS). But unlike WS, position can be set form one place.